ws-test-harness/core/src/main/java/com/sun/xml/ws/test/container/AbstractApplicationContainer.java

```diff
@@ -133,22 +134,30 @@ protected void updateWsgenOpts(TestService service) {
     protected void updateWsitClient(WAR war, DeployedService deployedService, String id) throws Exception {
         File wsitClientFile = new File(deployedService.getResources(), "wsit-client.xml");
         if (wsitClientFile.exists()) {
-            SAXReader reader = new SAXReader();
-            Document document = reader.read(wsitClientFile);
-            Element root = document.getRootElement();
-            Element policy = root.element("Policy");
-            Element sts = policy.element("ExactlyOne").element("All").element("PreconfiguredSTS");
+            Document document = XMLUtil.readXML(wsitClientFile, null);
+            Element root = document.getDocumentElement();
+            Element sts = XMLUtil.getElements(root, "//*[local-name()='Policy']/*[local-name()='ExactlyOne']/*[local-name()='All']/*[local-name()='PreconfiguredSTS']").get(0);
 
-            Attribute endpoint = sts.attribute("endpoint");
+            Attr  endpoint = sts.getAttributeNode("endpoint");
             endpoint.setValue(id);
 
-            Attribute wsdlLoc = sts.attribute("wsdlLocation");
-            String x = deployedService.service.wsdl.get(0).wsdlFile.toURI().toString();
-            wsdlLoc.setValue(x);
-
-            XMLWriter writer = new XMLWriter(new FileWriter(wsitClientFile));
-            writer.write(document);
-            writer.close();
+            Attr wsdlLoc = sts.getAttributeNode("wsdlLocation");
+            wsdlLoc.setValue(deployedService.service.wsdl.get(0).wsdlFile.toURI().toString());
+
+//            for (Element keystore : XMLUtil.getElements(root, "//*[local-name()='KeyStore']")) {
```

Comment:
Can't you just delete the commented out code?

Reply:
no I can't. At least not yet. While it is unused for metro-jaxws; metro-wsit will with quite high probability rely on this code, so this can be either removed or uncommented as soon as e2e tests in metro-wsit project pass